### DISCUSSION TITLE
Fresh Install Feedback

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#557] [Q&A] [answered]
As ~~threatened~~ promised, I used my spare phone (a Samsung S9) to install a new instance of Cruiser and go through with "new user eyes" so that I could make some notes for future Wiki articles.

What follows are my observations, not just in getting the default install to my liking, but some points to either discuss/improve/clarify.

1. Location permission request did not appear until I pressed "location" button
*Some apps try to do this immediately for you, prompting the permission request*
2. Some useful BRouter settings (eg: road avoidances) unavailable unless "Extended menu" enabled
*There's been talk of what stuff gets hidden behind the "extended" setting, but it was a surprise to find BRouter so "basic"!*
3. "Download offline map" popup bubble - expected to be clickable
*A useful alert, but may benefit from being clicked to invoke the Map manager. Great Britain is a 1.2GB download - given the size/wait, it would be handy a time/place to offer a 3-5 page slideshow of hints if this is the *first* map being downloaded.*
4. Route duration defaults to "Average", should it be "Calculated"?
*Given that the user would need to set their low/average/high speeds, "Calculated" seems the better option given its reliability*
5. Route diagram - consider use of the word "elevation"
6. "Keep screen on" not enabled for navigation
7. "Auto Zoom" not enabled for navigation
8. "Arrows shows route direction" not enabled
9. "Name of next Waypoint" panel not enabled
10. Navigation does not default to 3D view
*I'm sure this is the preferred view for most?*

This are the points exactly as I typed them in my text editor - so they are deliberately "raw".  As a fairly experienced user, I obviously had no trouble getting things set to my own tastes (without resorting to a profile import).  I found that it's far too easy to just rely on muscle-memory when you have the app showing the map style, app theme and layout *exactly* to your liking, so doing a fresh install with a fresh perspective can often clean-up little things sometimes missed.

Do any of the above need a topic/discussion about how/why/when etc?

**Comment by devemux86** [comment, marked as answer]
@MotoUKRider thanks for the detailed report!

> Location permission request did not appear until I pressed "location" button

All permissions are asked when they are needed (these are also the official Android instructions).
Isn't it annoying to start an app and it immediately asks for location, notification, etc. permissions?

> Some useful BRouter settings (eg: road avoidances) unavailable unless "Extended menu" enabled

The avoid options are enabled in each routing profile (fastest, curvy, extra curvy, etc).
Their levels are hidden to not confuse the not advanced users with too many options?

> "Download offline map" popup bubble - expected to be clickable

Map downloads will be improved in the future.

> Route duration defaults to "Average", should it be "Calculated"?

Thanks, this will be fixed. :+1: 

> Route diagram - consider use of the word "elevation"

It will be enriched in the future with more content.

> "Keep screen on" not enabled for navigation

Not sure, do we need to change the default here?

> "Auto Zoom" not enabled for navigation

I cannot confirm this?

> "Arrows shows route direction" not enabled

This is a user preference.

> "Name of next Waypoint" panel not enabled

Also a user preference what panels to enable.

> Navigation does not default to 3D view

This is also a user preference.
It is more for effect and consumes more battery.

**Comment by MotoUKRider** [comment]
@devemux86 A few comments back...

"Permissions" - When would Cruiser ask if it's OK to send me a notification?  Just before it tries to notify me? (ie: on my 1st route?)
I recall several other apps all asking me right from the outset. If a default install setting requires a permission, it may be wise to "ask" sooner rather than later.

"Keep screen on" - There are those that run "voice-only", so I can see that. But there's no "Android Auto", so I would argue that the majority of users are keen on a map that remains visible during navigation.

A pet-peeve, mentioned a while ago elsewhere, is the default "visible" on fuel-stations in the Layers menu. I assume this is down to handling of the settings?  It's mildly frustrating that a quick theme switch requires me to disable something Cruiser *thinks* I want to see. That's why I was pleased my "Day/Night Profile" tweak didn't suffer the same :smile_cat: 

Many others are personal settings, as you said.
It was a surprise just how "different" a fresh install of Cruiser is!

**Reply by devemux86** [reply]
Notifications are asked when they are set to be displayed by the user, e.g. in navigation.

About the map layers, they depend on the theme, if we unload a theme they are reset.
We may see how this can be improved in the future, as each theme has different layers.
